pkg/registry/batch/job/storage/storage_test.go

```diff
@@ -132,12 +135,117 @@ func TestUpdate(t *testing.T) {
 	)
 }
 
-func TestDelete(t *testing.T) {
-	storage, server := newStorage(t)
-	defer server.Terminate(t)
-	defer storage.Job.Store.DestroyFunc()
-	test := genericregistrytest.New(t, storage.Job.Store)
-	test.TestDelete(validNewJob())
+func TestJobDeletion(t *testing.T) {
+	ctx := genericapirequest.NewDefaultContext()
+	job := validNewJob()
+	key := "/jobs/" + metav1.NamespaceDefault + "/foo"
+	orphanDependents := true
+	orphanDeletionPropagation := metav1.DeletePropagationOrphan
+	backgroundDeletionPropagation := metav1.DeletePropagationBackground
+	tests := []struct {
```

Comment:
add test cases that invoke with requestinfo in the context for batch/v1 (warnings expected), batch/v2 (no warnings expected) and no API version info (no warnings expected)

Reply:
Done. PTAL and let me know if I missed any cases.